mPower2/mPower2/TaskGroupScheduleManager.swift

```diff
@@ -308,6 +324,28 @@ public class TaskGroupScheduleManager : ActivityGroupScheduleManager {
         let tasks: [RSDIdentifier] = [.medicationTask, .tremorTask, .tappingTask, .walkAndBalanceTask]
         return tasks.map { ReportQuery(reportKey: $0, queryType: .mostRecent, dateRange: nil) }
     }
+    
+    /// Returns the birth year and sex answer results ready to be added as previous results to a task
+    func createCRFTaskData() -> SBAReport? {
+        // Must have both birthYear and sex to create the CRF task data
+        if let yearAnswerStr = SBAProfileDataSourceObject.shared.profileTableItem(at: ProfileTableViewController.birthYearIndexPath)?.detail,
+           let yearInt = Int(yearAnswerStr),
+           let sexAnswerStr = SBAProfileDataSourceObject.shared.profileTableItem(at: ProfileTableViewController.sexIndexPath)?.detail {
+            
+            let json = [CRFDemographicsKeys.sex.stringValue : sexAnswerStr,
+                        CRFDemographicsKeys.birthYear.stringValue : yearInt] as RSDJSONSerializable
+            
+            return SBAReport(identifier: RSDIdentifier.heartSnapshotTask.identifier, date: Date(), json: json)
```

Comment:
Aren't you also suppose to include the VO2 max?

Reply:
Yeah this is really confusing, I'm going to put in some comments.

The VO2 max is included in the actual report generated and is what the HistoryDataManager uses to display the heart snapshot history item.  Because the TaskGroupScheduleManager only uses the heart snapshot report in the context of pushing in TaskData, and works.  I admit this is a pretty fragile solution.

I didn't like the approach either, but it was the only way I could figure out how to pass in the RSDTaskData.  
The real issue is that this function below, previousTaskData, in SBAReportManager is _public_ and not _open_.  If it was open, it would be a much more appropriate place to override.  I went with overriding what the function used - the report(for...) function instead versus trying to cherry pick what version MP2 is pointed at and make a hot fix like we were doing for CRF.

  ```
  public func previousTaskData(for taskIdentifier: RSDIdentifier) -> RSDTaskData? {
        do {
            if let report = report(with: taskIdentifier.stringValue) {
                return report
            }
            else {
                let sbbReport = try self.participantManager.getLatestCachedData(forReport: taskIdentifier.stringValue)
                let category = self.reportCategory(for: taskIdentifier.stringValue)
                return transformReportData(sbbReport, reportKey: taskIdentifier, category: category)
            }
        }
        catch let err {
            print("WARNING! Failed to query report. \(err)")
        }
        return nil
    }
```

Reply:
If you look at the structure of this code, there's probably a reason why I made it public rather than open. as in "if the report isn't found then you need to look in the cache. This is the open one. Override that.

```
    /// Find the most recent report for this activity identifier.
    ///
    /// - parameter activityIdentifier: The activity identifier for the client data associated with this task.
    /// - returns: The latest report (if any) associated with this activity identifier.
    open func report(with activityIdentifier: String) -> SBAReport? {
        let reportIdentifier = self.reportIdentifier(for: activityIdentifier)
        let report = reports.sorted { (lhs, rhs) -> Bool in
            return lhs.date < rhs.date
            }.last { $0.reportKey == reportIdentifier }
        return report
    }
```

Reply:
I don't get why you want to override that. In fact, you should *not* override this and instead override where you were doing so. This is public (and not open) to because it should check the cache if the report isn't found. I didn't understand that you were adding this to "fake" the report because you have code elsewhere to set it using the previous results.

Reply:
The motivation was mostly desperation, as I needed to find a way to use the results of the demographics survey to provide gender/birthYear to the heart snapshot task.  I'll keep it as is then.